staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go

```diff
@@ -816,10 +816,23 @@ func (q *Quantity) Value() int64 {
 
 // MilliValue returns the value of ceil(q * 1000); this could overflow an int64;
 // if that's a concern, call Value() first to verify the number is small enough.
+//
+// Deprecated: Prefer CheckedMilliValue which will error on overflow instead of
+// silently overflowing.
 func (q *Quantity) MilliValue() int64 {
 	return q.ScaledValue(Milli)
 }
 
+// CheckedMilliValue returns the value of ceil(q * 1000);
+// This may not fit in an int64 this could overflow an int64, in which case an error will be returned.
+func (q *Quantity) CheckedMilliValue() (int64, error) {
+	// TODO: check underflow?
+	if q.Value() > MaxMilliValue {
+		return 0, fmt.Errorf("value %v overflows MilliValue, cannot convert", q.Value())
```

Comment:
not sure if this is a sufficiently canonical error approach here

this package uses a number of named errors, but I figured you'd probably want the actual value that overflowed in the error ...